misaligned.c

```diff
@@ -1,21 +1,61 @@
 #include <stdio.h>
 #include <assert.h>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+enum MajorColor {WHITE, RED, BLACK, YELLOW, VIOLET};
+enum MinorColor {BLUE, ORANGE, GREEN, BROWN, SLATE};
+
+const char* majorColorNames[] = {"White", "Red", "Black", "Yellow", "Violet"};
+const char* minorColorNames[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+int numberOfMajorColors =
+    sizeof(majorColorNames) / sizeof(majorColorNames[0]);
+
+int numberOfMinorColors =
+    sizeof(minorColorNames) / sizeof(minorColorNames[0]);
+
+typedef struct {
+    enum MajorColor majorColorValue;
+    enum MinorColor minorColorValue;
+} ColorPair;
+
+
+
+int printColorMap() {    
     int i = 0, j = 0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+            printf("%d | %s | %s\n", ((i*5)+j+1), majorColorNames[i], minorColorNames[j]);
         }
     }
     return i * j;
 }
 
+
+ColorPair GetColorFromPairNumber(int pairNumber) {
+    ColorPair colorPair;
+    int zeroBasedPairNumber = pairNumber - 1;
+    colorPair.majorColorValue = 
+        (enum MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+    colorPair.minorColorValue =
+        (enum MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+    return colorPair;
+}
+
+void testNumberToPair(int pairNumber,
+    enum MajorColor expectedMajor,
+    enum MinorColor expectedMinor)
+{
+    ColorPair colorPair = GetColorFromPairNumber(pairNumber);
+    assert(colorPair.minorColorValue == expectedMinor);
+    assert(colorPair.majorColorValue == expectedMajor);
+}
+
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
+    printColorMap();

```

Comment:
The alignment isn't tested? It's about checking if the column separator | appears in the same position in every line. To ensure this, you can separate the formatting from the printing, or simply use an off-the-shelf alignment mechanism.
E.g., using a comma instead of | will make it import-able in excel, which will automatically align everything.